frontend/src/common/components/ConfirmDialogModal.tsx

```diff
@@ -0,0 +1,144 @@
+import React, {useState} from "react";
+
+import {SerializedError} from "@reduxjs/toolkit";
+import {FetchBaseQueryError} from "@reduxjs/toolkit/query";
+import {Button, Dialog} from "hds-react";
+import {Link} from "react-router-dom";
+
+import {NavigateBackButton, QueryStateHandler} from "./index";
+
+interface ConfirmDialogModalProps {
+    data;
+    modalText: string;
+    successText: string;
+    error: FetchBaseQueryError | SerializedError | undefined;
+    linkURL?: string;
+    linkText?: string;
+    buttonText?: string;
+    isLoading: boolean;
+    isVisible: boolean;
+    setIsVisible;
+    confirmAction;
+    cancelAction;
+}
+
+const ActionFailed = ({error, setIsVisible}) => {
+    const nonFieldError = ((error as FetchBaseQueryError)?.data as {message?: string})?.message || "";
+    return (
+        <>
+            <Dialog.Content>
+                <p>Virhe: {(error as FetchBaseQueryError)?.status}</p>
+                <p>{nonFieldError}</p>
+            </Dialog.Content>
+            <Dialog.ActionButtons>
+                <Button
+                    onClick={() => setIsVisible(false)}
+                    variant="secondary"
+                    theme="black"
+                >
+                    Sulje
+                </Button>
+            </Dialog.ActionButtons>
+        </>
+    );
+};
+
+const ActionSuccess = ({successText, linkURL, linkText}) => {
+    return (
+        <>
+            <Dialog.Content>{successText}</Dialog.Content>
+            <Dialog.ActionButtons>
+                <>
+                    <Link to={linkURL}>
+                        <Button
+                            variant="secondary"
+                            theme="black"
+                        >
+                            {linkText}
+                        </Button>
+                    </Link>
+                    <NavigateBackButton />
+                </>
+            </Dialog.ActionButtons>
+        </>
+    );
+};
+
+const ConfirmDialogModal = ({
+    data,
+    modalText,
+    successText,
+    linkURL,
+    linkText,
+    buttonText,
+    error,
+    isLoading,
+    isVisible,
+    setIsVisible,
+    confirmAction,
+    cancelAction,
+}: ConfirmDialogModalProps) => {
+    const [isConfirmed, setIsConfirmed] = useState(false);
+    return (
+        <Dialog
+            id="confirmation-modal"
+            closeButtonLabelText="args.closeButtonLabelText"
+            aria-labelledby="confirm-modal"
+            isOpen={isVisible}
+            close={() => setIsVisible(false)}
+            boxShadow
+        >
+            <Dialog.Header
+                id="confirmation-modal__header"
+                title="Vahvista toiminto"
+            />
+            {isConfirmed && error ? (
```

Comment:
I don't see the point in this variable. If error exists, it means that the 'confirm' button has been already pressed 🤔 
This could be removed without it making any changes to functionality

Reply:
Yup - you're right. It's a remnant from when I was thinking that the confirmation window would itself show a message after a successful confirmed action. But as the toast notification is used for this, it is unnecessary.